### PR TITLE
Move documents into a separate store object

### PR DESCRIPTION
**Pull request by tejans24**
connects #2829 

* This separates the documents from the reader reducer.
* Changes the selectors to take in the entire state and filter based on the objects in the store.

### CODE REVIEW

File: client/app/reader/DocumentList/DocumentsReducer.js [hunk 1]
```diff
@@ -150,6 +150,24 @@ const documentsReducer = (state = {}, action = {}) => {
         }
       }
     });
+  case Constants.REQUEST_REMOVE_TAG_FAILURE:
```
Comment (mdbenjam) [inline client/app/reader/DocumentList/DocumentsReducer.js]: Was this broken before?

File: client/app/reader/reducer.js [hunk 2]
```diff
@@ -337,45 +335,10 @@ const reducer = (state = {}, action = {}) => {
...
-            }
-          }
-        }
-      }
-    });
   case Constants.SCROLL_TO_COMMENT:
     return update(state, {
       ui: { pdf: { scrollToComment: { $set: action.payload.scrollToComment } } }
     });
-  case Constants.TOGGLE_COMMENT_LIST:
```
[the remark this answers is not in the document]
Comment (tejans24) [reply client/app/reader/reducer.js]: Yes, this was not removed when things were moved to a different file and it should have been.

File: client/app/reader/DocumentList/DocumentsReducer.js (hunk 1 above)
Comment (tejans24) [reply client/app/reader/DocumentList/DocumentsReducer.js]: The documents object was still part of the readerReducer, now it's a separate object in the Store. I'm guessing it wasn't broken and it was in the wrong place.

File: client/test/node/reader/DecisionReviewer-test.js [hunk 3]
```diff
@@ -87,75 +87,6 @@ const getWrapper = (store) => mount(
 /* eslint-disable max-statements */
 describe('DecisionReviewer', () => {
   let wrapper;
-  let setUpDocuments;
-
-  beforeEach(() => {
```
Comment (mdbenjam) [inline client/test/node/reader/DecisionReviewer-test.js]: Is all this section just duplicated?

**Pull request merged**